src/element-style-observer.js

```diff
@@ -274,11 +274,85 @@ export default class ElementStyleObserver {
 		// transition: all, var(--style-observer-transition, all);
 		// so we can't just concatenate with whatever the existing value is
 		const prefix = transition ? transition + ", " : "";
-		this.target.style.setProperty("transition", prefix + sot);
+		this.setProperty("transition", prefix + sot);
 
 		this.updateTransitionProperties();
 	}
 
+	/**
+	 * Whether the target has an open shadow root (and the modern adoptedStyleSheets API is supported).
+	 * @type { boolean }
+	 * @private
+	 */
+	get _isHost () {
+		return (
+			this.target.shadowRoot && !Object.isFrozen(this.target.shadowRoot.adoptedStyleSheets)
+		);
+	}
+
+	/**
+	 * Shadow style sheet. Only used if _isHost is true.
+	 * @type { CSSStyleSheet | undefined }
+	 * @private
+	 */
+	_shadowSheet;
+
+	/**
+	 * Any styles we've set on the target, for any reason.
+	 * @type { Record<string, string> }
+	 * @private
+	 */
+	_styles = {};
+
+	/**
+	 * Set a CSS property on the target.
+	 * @param {string} property
+	 * @param {string} value
+	 * @return {void}
+	 */
+	setProperty (property, value) {
+		let inlineStyle = this.target.style;
+		let style = inlineStyle;
+		if (this._isHost) {
+			// This has an open shadow root.
+			// We can use an adopted shadow style to avoid manipulating its style attribute
+			if (!this._shadowSheet) {
+				this._shadowSheet = new CSSStyleSheet();
+				this._shadowSheet.insertRule(`:host { }`);
```

Comment:
Is this rule subject to change? I don't fully understand its point. Or does it allow us to work with `_shadowSheet` the way we do in the following lines?

Other than that question, LGTM!

Reply:
It creates the rule that we reference lower down (in line 338)